docs/docs/troubleshooting/hardware-issues.mdx

```diff
@@ -0,0 +1,183 @@
+      { label: "Seeed Studio XIAO Compatible Board", value: "xiao" },
+    ]}
+  >
+    {/* eslint-disable-next-line */}
+    {props.children}
+  </Tabs>
+
+);
+
+export const PinTabs = (props) => (
+    <Tabs
+    groupId="pin-approaches"
+    defaultValue="shield"
+    values={[
+      { label: "Tester Shield", value: "shield" },
+      { label: "Multimeter", value: "multimeter" },
+      { label: "Visual Inspection", value: "visual" },
+    ]}
+  >
+    {/* eslint-disable-next-line */}
+    {props.children}
+  </Tabs>
+
+);
+
+## Pin Connections
+
+### Identifying Issues
+
+There are three general approaches we recommend for identifying potential issues with microcontroller pins.
+
+<PinTabs>
+<TabItem value="shield">
+ZMK comes with built-in "tester" shields for [pro micro combatible boards](https://cdn.sparkfun.com/assets/9/c/3/c/4/523a1765757b7f5c6e8b4567.png) and [Seeed Studio XIAO combatible boards](https://files.seeedstudio.com/wiki/Seeeduino-XIAO/img/Seeeduino-XIAO-pinout-1.jpg). The shields allow for easy testing of GPIO pins, but cannot be used to troubleshoot power pins.
+
+#### Acquiring a tester uf2
+
+<TesterTabs>
+<TabItem value="promicro">
+Find the `build.yaml` file in your `zmk-config` folder and add an additional `tester_pro_micro` build for your board:
+
+```yml title="build.yaml"
+include:
+  # -------------------
+  # Your keyboards here
+  # -------------------
+  - board: nice_nano_v2 # Replace with your board of use
+    shield: tester_pro_micro
+```
+
+Save the file, commit the changes and push them to GitHub. Download the new firmware zip file build by the latest GitHub Actions job. In it you will find an additional `tester_pro_micro` UF2 image file. Flash this file to your board.
+
+</TabItem>
+<TabItem value="xiao">
+Find the `build.yaml` file in your `zmk-config` folder and add an additional `tester_xiao` build for your board:
+
+```yml title="build.yaml"
+include:
+  # -------------------
+  # Your keyboards here
+  # -------------------
+  - board: nice_nano_v2 # Replace with your board of use
+    shield: tester_xiao
+```
+
+Save the file, commit the changes and push them to GitHub. Download the new firmware zip file build by the latest GitHub Actions job. In it you will find an additional `tester_xiao` UF2 image file. Flash this file to your board.
+
+</TabItem>
+</TesterTabs>
+After flashing, open a text editor of your choice. Shorting a GPIO pin to ground (with e.g. a wire or tweezers, hold one end go ground and tap the pin with the other end) will now trigger the keyboard to send an informative message on which pins were triggered.
+- If one pin was triggered, then the pin is working fine.
+- If two were triggered at once, then these pins are shorted together on your device.
+- If no pin was triggered, try shorting a working pin to the pin you were trying to trigger to test if it is shorted to ground on your device. Otherwise, the pin is disconnected or broken.
```

Comment:
I feel like this is not complete as a tab, e.g. once you identify one of the above two cases then you'd then want to use the multimeter to figure out where exactly the issue is.

As such, this tab might be better off as a subsection after the other two as a helper tool that can be used in conjunction with e.g. a multimeter.

Reply:
I feel like all three rely on each other for various things. I've left them as tabs, but added some references to the other methods in each tab.